package/MDAnalysis/coordinates/LAMMPS.py

```diff
@@ -552,16 +582,46 @@ def _read_next_timestep(self):
 
         indices = np.zeros(self.n_atoms, dtype=int)
 
-        f.readline()  # ITEM ATOMS etc
-        for i in range(self.n_atoms):
-            idx, _, xs, ys, zs = f.readline().split()
+        atomline = f.readline()  # ITEM ATOMS etc
+        attrs = atomline.split()[2:]  # attributes on coordinate line
+        col_ids = {attr: i for i, attr in enumerate(attrs)}  # column numbers
+        keys = col_ids.keys()
+        # check for ids and what type of coordinate convention
+        ids = "id" in keys
+        # coord_dict keys = convention keys = column data or False
+        coord_dict = {conv: False for conv in self._conventions[1:]}
+        for coordtype, col_names in self._coordtype_column_names.items():
+            if all(axis in col_ids for axis in col_names):
```

Comment:
What happens if it's partially missing? Do we just ignore those entries? Should we be raising an error or is the standard allowing for this?

Reply:
Can the order be reversed or badly arranged? What do we do if it's y/z/x instead of x/y/z?

Reply:
The column indices means that the order is agnostic. What this code does is:
* ignore the convention if it's partially missing (e.g. only y and z are provided)
* store the order of the columns in `coord_dict`
* which then gets moved to `coord_cols`.

Reply:
That sounds sensible then.